lms/djangoapps/django_comment_client/utils.py

```diff
@@ -131,13 +131,37 @@ def get_accessible_discussion_xblocks_by_course_id(course_id, user, include_all=
     are accessible to the given user.
     """
     all_xblocks = modulestore().get_items(course_id, qualifiers={'category': 'discussion'}, include_orphans=False)
+    if not all_xblocks:
+        return []
 
+    prefetched_group_data = prefetch_group_data(user, all_xblocks[0])
```

Comment:
Is it worth putting this in a cache?  If we somehow make it to the `else` portion of `_has_group_access()` above, it means we didn't have prefetched data when loading the discussion tab, and it's likely that we again won't have it if the user navigates away and comes back.  The last line of that `else` block could then cache the `user_groups` and `partition_groups` that it just worked so hard to compute.  Does that make sense?